src/charset.c

```diff
@@ -527,26 +527,31 @@ transchar_buf(buf_T *buf, int c)
  */
     char_u *
 transchar_byte(int c)
+{
+    return transchar_byte_buf(curbuf, c);
+}
+
+    char_u *
+transchar_byte_buf(buf_T *buf,int c)
 {
     if (enc_utf8 && c >= 0x80)
     {
-	transchar_nonprint(curbuf, transchar_charbuf, c);
-	return transchar_charbuf;
+        transchar_nonprint(buf, transchar_charbuf, c);
+        return transchar_charbuf;
     }
-    return transchar(c);
+    return transchar_buf(buf, c);
 }
-
 /*
  * Convert non-printable character to two or more printable characters in
- * "charbuf[]".  "charbuf" needs to be able to hold five bytes.
+ * "buf[]".  "charbuf" needs to be able to hold five bytes.
  * Does NOT work for multi-byte characters, c must be <= 255.
  */
     void
 transchar_nonprint(buf_T *buf, char_u *charbuf, int c)
 {
     if (c == NL)
 	c = NUL;		// we use newline in place of a NUL
-    else if (c == CAR && get_fileformat(buf) == EOL_MAC)
+    else if (buf != NULL && c == CAR && get_fileformat(buf) == EOL_MAC)
 	c = NL;			// we use CR in place of  NL in this case
```

Comment:
```diff
-	c = NL;			// we use CR in place of  NL in this case
+	c = NL;			// we use CR in place of NL in this case
```
This is not your change, but there is an extra space in the comment.

Reply:
thank you for the suggestions, I have commited them all